src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,54 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.state.ProgramState;
+import duke.task.Deadline;
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * Represents a command to add a task with a deadline.
+ */
+public class DeadlineCommand extends Command {
+    private String taskDescription;
+    private String dueTime;
+
+    /**
+     * Constructs a new deadline command.
+     *
+     * @param body The body of the command.
+     */
+    public DeadlineCommand(String body) {
+        super(body);
+        String[] parts = body.split("/by", 2);
+        this.taskDescription = parts[0].trim();
+        this.dueTime = parts.length > 1 ? parts[1].trim() : "";
+    }
+
+    /**
+     * Executes the command. This command adds a task with a deadline to the task
+     * list. The program state is set to normal after the command is executed, even
```

Comment:
I like that your java documentation is written well and explains the function usage well.